### PR TITLE
xds/bootstrap: refactor to support top level and per-authority server config

### PR DESCRIPTION
**Pull request by menghanl**
Changes
- new struct for server config (URI, creds, v2/v3, node), and reused for top level and per-authority server
  - and fix all the usages
- logging for unknown fields
- node proto message is now for each server, to be consistent with TransportAPI version
- config test to use cmp.Diff

RELEASE NOTES: N/A

### CODE REVIEW

File: xds/internal/xdsclient/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -342,6 +342,8 @@ func NewConfigFromContents(data []byte) (*Config, error) {
 			if err := json.Unmarshal(v, &config.Authorities); err != nil {
 				return nil, fmt.Errorf("xds: json.Unmarshal(%v) for field %q failed during bootstrap: %v", string(v), k, err)
 			}
+		default:
+			logger.Infof("Bootstrap content has unknown field: %s", k)
```
Comment (easwars) [inline xds/internal/xdsclient/bootstrap/bootstrap.go:340]: Should this be at warning level?

File: xds/internal/xdsclient/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -116,6 +116,7 @@ func (sc *ServerConfig) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// Authority represents an Authority for an xDS control plane server.
```
Comment (easwars) [inline xds/internal/xdsclient/bootstrap/bootstrap.go:119]: Can we have a comment which is more helpful than this :)

**Review by easwars — APPROVED** [review]

File: xds/internal/xdsclient/bootstrap/bootstrap.go (hunk 2 above)
Comment (menghanl) [reply xds/internal/xdsclient/bootstrap/bootstrap.go]: I cannot think of useful comment for this struct. It's usage is explained in the Config struct where it's used.

I added a sentence.

File: xds/internal/xdsclient/bootstrap/bootstrap.go (hunk 1 above)
Comment (menghanl) [reply xds/internal/xdsclient/bootstrap/bootstrap.go]: Changed. I think it's lower than warning.

**Pull request merged**